src/mbgl/map/map.cpp

```diff
@@ -1095,11 +1099,11 @@ void Map::Impl::onUpdate(Update flags) {
 }
 
 void Map::Impl::onStyleLoaded() {
-    backend.notifyMapChange(MapChangeDidFinishLoadingStyle);
+    observer.onDidFinishLoadingStyle();
 }
 
 void Map::Impl::onStyleError() {
-    backend.notifyMapChange(MapChangeDidFailLoadingMap);
+    observer.onDidFailLoadingMap();
```

Comment:
We should plumb [`res.error`](https://github.com/mapbox/mapbox-gl-native/blob/936cc33d5291a77bc4e0dda9c7956860bef5f872/src/mbgl/map/map.cpp#L371) through to here or implement `onResourceError()` instead to allow the SDKs to provide more information about the failure. Currently, in order to fudge parity with MapKit, the iOS and macOS SDKs have to [synthesize an uninformative NSError](https://github.com/mapbox/mapbox-gl-native/blob/de9c0748a216d05b15509f1a73086f70067643b2/platform/ios/src/MGLMapView.mm#L4830), but `res.error`’s `message` field would be perfect for this purpose.